drivers/wifi/simplelink/simplelink_sockets.c

```diff
@@ -283,6 +283,17 @@ static int simplelink_connect(int sd, const struct sockaddr *addr,
 		retval = 0;
 	}
 
+	/* Warn users when root CA is not in the certificate catalog.
+	 * For enhanced security, users should update the catalog with the
+	 * certificates for sites the device is expected to connect to. Note
+	 * the connection is established successfully even when the root CA
+	 * is not part of the catalog.
+	 */
+	if (retval == SL_ERROR_BSD_ESECUNKNOWNROOTCA) {
+		LOG_WRN("Unknown root CA used");
```

Comment:
@pfalson I didn't want to make this into an error, because it'd impose work on someone who just wish to do development or try out an example using a test certificate they have created themselves.  They'd have to figure out how to add it to the certificate catalog - which is throw-away work at the end of the day if their intent is to just use an existing root CA in their production system. We have taken the same approach in the past in TI software for ease of use.

Maybe I can make the wording of the warning stronger? "Unknown root CA used. For proper security, please use a root CA that is part of the certificate catalog in production systems."

Reply:
> Maybe I can make the wording of the warning stronger? 

Well, I actually was talking about replacing LOG_WRN() with LOG_ERR() either, not stopping application here or something. And it's mostly about "thinking about security" vs getting into situation when something was done wrong and nobody even paid attention. I developer myself, and this is merely a sample, so I very easily buy your arguments, and happy with it staying as is ;-). The whole point is again to make sure it was considered - sooner or later someone will catch a real security issue somewhere. Thanks.

Reply:
I have decided to leave it as LOG_WRN to be consistent, given a few lines above we treated a similar certificate issue in the same way.